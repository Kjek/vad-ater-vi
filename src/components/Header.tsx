import GitHubButton from '@component/GitHubButton';
import SiteLogo from './SiteLogo';
import ThemeButton from './ThemeButton';

const Header = () => {
  return (
    <header>
      <nav className='flex justify-between bg-neutral-100 py-2 px-4 shadow-md transition duration-500 dark:border-spacing-1 dark:border-b dark:border-gray-700 dark:bg-gray-900 dark:shadow-none'>
        <div className='m-auto flex w-full grow items-center justify-center gap-4 sm:ml-20 '>
          <SiteLogo />
          <h1 className='text-3xl font-bold text-gray-800 dark:text-gray-300'>
            Lunch i Sundsvall
          </h1>
        </div>
        <div className='flex items-center gap-2'>
          <div className='rounded-md'>
            <ThemeButton />
          </div>
          <div className='rounded-md'>
            <GitHubButton />
          </div>
        </div>
      </nav>
    </header>
  );
};

export default Header;
