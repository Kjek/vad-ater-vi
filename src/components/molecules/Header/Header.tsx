import GitHubButton from '@component/atoms/GitHubButton/GitHubButton';
import SiteLogo from '@component/atoms/SiteLogo/SiteLogo';
import Text from '@component/atoms/Text/Text';
import ThemeButtonSkeleton from '@component/atoms/ThemeButton/ThemeButtonSkeleton';
import dynamic from 'next/dynamic';

const ThemeButton = dynamic(
  () => import('../../atoms/ThemeButton/ThemeButton'),
  {
    ssr: false,
    // Make sure to code a placeholder so the UI doesn't jump when the component loads
    loading: () => <ThemeButtonSkeleton />,
  }
);

const Header = () => {
  return (
    <header>
      <nav className='flex justify-between bg-neutral-100 px-4 py-2 shadow-md transition duration-500 dark:border-spacing-1 dark:border-b dark:border-gray-700 dark:bg-gray-900 dark:shadow-none'>
        <div className='m-auto flex w-full grow items-center justify-center gap-4 sm:ml-20 '>
          <SiteLogo />
          <Text variant='h1'>Lunch i Sundsvall</Text>
        </div>
        <div className='flex items-center gap-2'>
          <div className='rounded-md'>
            <ThemeButton />
          </div>
          <div className='rounded-md'>
            <GitHubButton />
          </div>
        </div>
      </nav>
    </header>
  );
};

export default Header;
