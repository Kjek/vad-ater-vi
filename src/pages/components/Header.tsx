import GitHubIcon from '~/assets/icons/GitHub-Icon';
import Switcher from './Switcher';

const Header = () => {
  return (
    <header>
      <nav className='bg-gradient-to-r pb-1 dark:from-[#ef473a] dark:to-[#cb2d3e]'>
        <div className='flex h-12 justify-between dark:bg-neutral-800'>
          <div className='m-auto justify-center'>
            <h1 className='text-3xl font-bold text-black dark:text-white'>
              Lunch i Sundsvall
            </h1>
          </div>
          <div className='flex w-16 justify-end'>
            <Switcher />
          </div>
          <div className='flex w-16 justify-end'>
            <GitHubIcon className='w-16 p-1' />
          </div>
        </div>
      </nav>
    </header>
  );
};

export default Header;
