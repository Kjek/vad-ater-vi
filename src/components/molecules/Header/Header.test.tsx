import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Header from './Header';

describe(Header, () => {
  it('renders correctly', () => {
    const { container } = render(<Header />);
    expect(container).toBeInTheDocument();
    expect(container).toBeVisible();
  });

  it('matches snapshot', () => {
    const defaultRender = render(<Header />);
    expect(defaultRender).toMatchSnapshot();
  });
});
