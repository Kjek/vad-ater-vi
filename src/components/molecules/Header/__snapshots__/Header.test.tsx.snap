// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <nav
          class="flex justify-between bg-neutral-100 px-4 py-2 shadow-md transition duration-500 dark:border-spacing-1 dark:border-b dark:border-gray-700 dark:bg-gray-900 dark:shadow-none"
        >
          <div
            class="m-auto flex w-full grow items-center justify-center gap-4 sm:ml-20 "
          >
            <img
              alt="Lunch Image"
              class="h-fit"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fsite-image.png&w=96&q=75"
              srcset="/_next/image?url=%2Fsite-image.png&w=48&q=75 1x, /_next/image?url=%2Fsite-image.png&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <h1
              class="text-3xl font-bold text-gray-800 dark:text-gray-300"
            >
              Lunch i Sundsvall
            </h1>
          </div>
          <div
            class="flex items-center gap-2"
          >
            <div
              class="rounded-md"
            >
              <div
                class="flex animate-pulse rounded-md hover:ring-gray-300"
              >
                <span
                  class="cursor-pointer self-center rounded-md bg-gray-200 p-2 transition duration-500 dark:bg-gray-700"
                >
                  <div
                    class="size-6"
                  />
                </span>
              </div>
            </div>
            <div
              class="rounded-md"
            >
              <button
                class="flex rounded-md hover:ring-gray-300"
                type="button"
              >
                <a
                  class="cursor-pointer self-center rounded-md bg-white p-2 transition duration-500 hover:bg-gray-200 hover:transition-none dark:bg-gray-800 dark:hover:bg-gray-700"
                  href="https://github.com/Kjek/vad-ater-vi"
                  title="Länk till sidans GitHub repository"
                >
                  <svg
                    class="size-6 text-black dark:text-white"
                    fill="none"
                    height="15"
                    viewBox="0 0 15 15"
                    width="15"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M7.49933 0.25C3.49635 0.25 0.25 3.49593 0.25 7.50024C0.25 10.703 2.32715 13.4206 5.2081 14.3797C5.57084 14.446 5.70302 14.2222 5.70302 14.0299C5.70302 13.8576 5.69679 13.4019 5.69323 12.797C3.67661 13.235 3.25112 11.825 3.25112 11.825C2.92132 10.9874 2.44599 10.7644 2.44599 10.7644C1.78773 10.3149 2.49584 10.3238 2.49584 10.3238C3.22353 10.375 3.60629 11.0711 3.60629 11.0711C4.25298 12.1788 5.30335 11.8588 5.71638 11.6732C5.78225 11.205 5.96962 10.8854 6.17658 10.7043C4.56675 10.5209 2.87415 9.89918 2.87415 7.12104C2.87415 6.32925 3.15677 5.68257 3.62053 5.17563C3.54576 4.99226 3.29697 4.25521 3.69174 3.25691C3.69174 3.25691 4.30015 3.06196 5.68522 3.99973C6.26337 3.83906 6.8838 3.75895 7.50022 3.75583C8.1162 3.75895 8.73619 3.83906 9.31523 3.99973C10.6994 3.06196 11.3069 3.25691 11.3069 3.25691C11.7026 4.25521 11.4538 4.99226 11.3795 5.17563C11.8441 5.68257 12.1245 6.32925 12.1245 7.12104C12.1245 9.9063 10.4292 10.5192 8.81452 10.6985C9.07444 10.9224 9.30633 11.3648 9.30633 12.0413C9.30633 13.0102 9.29742 13.7922 9.29742 14.0299C9.29742 14.2239 9.42828 14.4496 9.79591 14.3788C12.6746 13.4179 14.75 10.7025 14.75 7.50024C14.75 3.49593 11.5036 0.25 7.49933 0.25Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </a>
              </button>
            </div>
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <nav
        class="flex justify-between bg-neutral-100 px-4 py-2 shadow-md transition duration-500 dark:border-spacing-1 dark:border-b dark:border-gray-700 dark:bg-gray-900 dark:shadow-none"
      >
        <div
          class="m-auto flex w-full grow items-center justify-center gap-4 sm:ml-20 "
        >
          <img
            alt="Lunch Image"
            class="h-fit"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fsite-image.png&w=96&q=75"
            srcset="/_next/image?url=%2Fsite-image.png&w=48&q=75 1x, /_next/image?url=%2Fsite-image.png&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h1
            class="text-3xl font-bold text-gray-800 dark:text-gray-300"
          >
            Lunch i Sundsvall
          </h1>
        </div>
        <div
          class="flex items-center gap-2"
        >
          <div
            class="rounded-md"
          >
            <div
              class="flex animate-pulse rounded-md hover:ring-gray-300"
            >
              <span
                class="cursor-pointer self-center rounded-md bg-gray-200 p-2 transition duration-500 dark:bg-gray-700"
              >
                <div
                  class="size-6"
                />
              </span>
            </div>
          </div>
          <div
            class="rounded-md"
          >
            <button
              class="flex rounded-md hover:ring-gray-300"
              type="button"
            >
              <a
                class="cursor-pointer self-center rounded-md bg-white p-2 transition duration-500 hover:bg-gray-200 hover:transition-none dark:bg-gray-800 dark:hover:bg-gray-700"
                href="https://github.com/Kjek/vad-ater-vi"
                title="Länk till sidans GitHub repository"
              >
                <svg
                  class="size-6 text-black dark:text-white"
                  fill="none"
                  height="15"
                  viewBox="0 0 15 15"
                  width="15"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M7.49933 0.25C3.49635 0.25 0.25 3.49593 0.25 7.50024C0.25 10.703 2.32715 13.4206 5.2081 14.3797C5.57084 14.446 5.70302 14.2222 5.70302 14.0299C5.70302 13.8576 5.69679 13.4019 5.69323 12.797C3.67661 13.235 3.25112 11.825 3.25112 11.825C2.92132 10.9874 2.44599 10.7644 2.44599 10.7644C1.78773 10.3149 2.49584 10.3238 2.49584 10.3238C3.22353 10.375 3.60629 11.0711 3.60629 11.0711C4.25298 12.1788 5.30335 11.8588 5.71638 11.6732C5.78225 11.205 5.96962 10.8854 6.17658 10.7043C4.56675 10.5209 2.87415 9.89918 2.87415 7.12104C2.87415 6.32925 3.15677 5.68257 3.62053 5.17563C3.54576 4.99226 3.29697 4.25521 3.69174 3.25691C3.69174 3.25691 4.30015 3.06196 5.68522 3.99973C6.26337 3.83906 6.8838 3.75895 7.50022 3.75583C8.1162 3.75895 8.73619 3.83906 9.31523 3.99973C10.6994 3.06196 11.3069 3.25691 11.3069 3.25691C11.7026 4.25521 11.4538 4.99226 11.3795 5.17563C11.8441 5.68257 12.1245 6.32925 12.1245 7.12104C12.1245 9.9063 10.4292 10.5192 8.81452 10.6985C9.07444 10.9224 9.30633 11.3648 9.30633 12.0413C9.30633 13.0102 9.29742 13.7922 9.29742 14.0299C9.29742 14.2239 9.42828 14.4496 9.79591 14.3788C12.6746 13.4179 14.75 10.7025 14.75 7.50024C14.75 3.49593 11.5036 0.25 7.49933 0.25Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </a>
            </button>
          </div>
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
